mat64/product.go

```diff
@@ -0,0 +1,190 @@
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package mat64
+
+import "fmt"
+
+// Product calculates the product of the given factors and places the result in
+// the receiver. The order of multiplication operations is optimized to minimize
```

Comment:
We don't have to lie, I was thinking of saying something like "Product selects the order of matrix multiplications and so in many cases will be faster than writing individual multiplies by hand". We can leave as is also, I don't feel that strongly.